modules/core/src/main/scala/gem/math/Angle.scala

```diff
@@ -119,54 +127,63 @@ object Angle {
+    val ms = (micros / (1000L))                    % 1000L
+    val s  = (micros / (1000L * 1000L))            % 60L
+    val m  = (micros / (1000L * 1000L * 60L))      % 60L
+    val d  = (micros / (1000L * 1000L * 60L * 60L))
+    (d.toInt, m.toInt, s.toInt, ms.toInt, µs.toInt)
   }
 
   /**
-   * Integral angle represented as a sum of degrees, arcminutes, arcseconds, and milliarcseconds.
-   * This type is exact and isomorphic to Angle.
+   * Integral angle represented as a sum of degrees, arcminutes, arcseconds, milliarcseconds and
+   * microarcseconds. This type is exact and isomorphic to Angle.
    */
   final case class DMS(toAngle: Angle) {
     val (
       degrees: Int,
       arcminutes: Int,
       arcseconds: Int,
-      milliarcseconds: Int
-    ) = Angle.toMillisexigesimal(toAngle.toMilliarcseconds)
+      milliarcseconds: Int,
+      microarcseconds: Int
+    ) = Angle.toMicrosexigesimal(toAngle.toMicroarcseconds)
     override final def toString =
-      f"$degrees:$arcminutes%02d:$arcseconds%02d.$milliarcseconds%03d"
+      f"$degrees:$arcminutes%02d:$arcseconds%02d.$milliarcseconds%03d$microarcseconds%03d"
   }
 
   /**
-   * Construct a new Angle of the given magnitide as a sum of degrees, arcminutes, arcseconds, and
-   * milliarcseconds. Exact modulo 360°.
+   * Construct a new Angle of the given magnitide as a sum of degrees, arcminutes, arcseconds,
+   * milliarcseconds, and microarcseconds. Exact modulo 360°.
    */
-  def fromDMS(degrees: Int, arcminutes: Int, arcseconds: Int, milliarcseconds: Int): Angle =
-    fromMilliarcseconds(
-      milliarcseconds.toLong +
-      arcseconds.toLong * 1000 +
-      arcminutes.toLong * 1000 * 60 +
-      degrees.toLong    * 1000 * 60 * 60
+  def fromDMS(
+    degrees:         Int,
+    arcminutes:      Int,
+    arcseconds:      Int,
+    milliarcseconds: Int,
+    microarcseconds: Int
+  ): Angle =
+    fromMicroarcseconds(
+      microarcseconds.toLong +
+      milliarcseconds.toLong * 1000 +
+      arcseconds.toLong      * 1000 * 1000 +
+      arcminutes.toLong      * 1000 * 1000 * 60 +
+      degrees.toLong         * 1000 * 1000 * 60 * 60
     )
 
 }
 
 
 
 /**
- * Exact hour angles represented as integral milliseconds. These values form an Abelian group over
+ * Exact hour angles represented as integral microseconds. These values form an Abelian group over
  * addition, where the inverse is reflection around the 0-12h axis. This is a subgroup of the
- * integral Angles where milliarcseconds are evenly divisible by 15.
+ * integral Angles where microarcseconds are evenly divisible by 15.
  */
-final class HourAngle private (mas: Int) extends Angle(mas) {
+final class HourAngle private (µas: Long) extends Angle(µas) {
 
   // Sanity checks … should be correct via the companion constructor.
-  assert(toMilliarcseconds %  15 === 0, s"Invariant violated. $mas isn't divisible by 15.")
+  assert(toMicroarcseconds %  15 === 0, s"Invariant violated. $µas isn't divisible by 15.")
```

Comment:
`apply` normalizes the angle to [0-360°) so it never fails. This assertion is just another check that `apply` is doing what it's supposed to do.